assets/activation-scripts/activate

```diff
@@ -156,15 +203,19 @@ if [ -n "$FLOX_CMD" ]; then
   set -- "$FLOX_CMD"
 fi
 
-# Set FLOX_ENV as the path by which all flox scripts can make reference to
-# the environment to which they belong. Use this to define the path to the
-# activation scripts directory.
-# TODO: reconcile with CLI which should be setting this. We must override
-#       the value coming from the CLI for now because it won't be set for
-#       container invocations, and it would have the incorrect value for
-#       nested flox activations.
-_FLOX_ENV="$($_dirname -- "${BASH_SOURCE[0]}")"
-export FLOX_ENV="${FLOX_ENV:-$_FLOX_ENV}"
+# Restore required variables that are exposed as publicly documented.
+export FLOX_ENV="${_FLOX_ENV}"
+
+# Restore optional variables that are exposed as publicly documented.
+# NB: `generate_*_start_commands()` performs the same logic except for zsh.
+for var_key in FLOX_ENV_CACHE FLOX_ENV_PROJECT FLOX_ENV_DESCRIPTION; do
+  eval "var_val=\${_$var_key-}"
+  if [ -n "$var_val" ]; then
+    export $var_key="${var_val}"
+  else
+    unset $var_key
+  fi
+done
```

Comment:
This could be a good candidate for a `flox-activations` sub-command since we do the same work in a few places and the shell script is hard to read, but I wanted to get a working fix in first before dealing with the boilerplate.